2/answers.txt

```diff
@@ -0,0 +1,231 @@
+
+Allowing you to compare and merge those changes individually and check for issues that may arise or errors that are made before combining them into the master branch.
+
+Q7: Why is it important to frequently commit and push your changes to a remote repository (eg your forked repo on GitHub)?
+
+Frequent commit and push will safeguard your work against loss of data if the original is somehow compromised, as well as ensuring that with smaller consistent changes, anything that causes
+a problem elsewhere or breaks something is easily identifiable, committing a lot of work at once increases the chance of a conflict with code that is submitted and will make it harder to
+troubleshoot an error should one arise.
+
+Q8: Why are VM snapshots important?
+
+VM Snapshots when done regularly will allow you to restore a VM to a state much closer to that which it was before an error or failure occurs, saving lots of time and effort in lost work
+Ultimately saving the state of the machine like this is similar to regularly saving your work, if it goes down and you have to restore it, you will have lost less if you snapshot regularly
+restoring it to a state before it experienced a problem.
+
+Q9: When and why would you choose to use incremental backups vs full backups? 
+
+Incremental backups are especially useful when backing up large amounts of data or data that changes infrequently, by having the backup only transfer that which has changed since the
+previous backup, you can save a lot of network traffic, data transfer costs.
+
+Q10: What port is SSH usually exposed on? What are the security implications of having this port exposed to the internet? Is it worthwhile changing the port?
+
+SSH is exposed on port 22 by default, the security implications are that an attacker would know where to look for the port making a connection atttempt, changing the port would likely make
+the default operation of a lot of programs function incorrectly without additional configuration, such as any SSH connection would need to be told explicitly which port is listening for the SSH commands.
+
+I would guess a determined attacker would scan all ports on a location for the port that responds to an SSH request if thats what they were looking for and perhaps the obfuscation of the SSH port would Ultimately
+prove pointless because of this? Ultimately you might end up with less risk because bots set to attempt port 22 would miss it but ultimately if its configured correctly such an attack would be futile.
+
+Q11: What could happen if your SSH key is compromised?
+
+If your private SSH key is compromised, then forward secrecy should only allow that existing session to be compromised, and may allow someone to access the server, but not the remote.
```

Comment:
Incorrect.

The remote is generally not considered to be the party at risk as it is not the one running the SSH server. The server has very limited ability to compromise the remote via the bi-directional connection.

Forward secrecy protects only past connections from being decrypted. it offers no protection for future connections.
If your keys are compromised, your server is compromised from that point onwards.

You would need to perform the following:
1. Revoke the old keys immediately.
2. Consider the server compromised. This is far more severe if the SSH user is able to elevate permissions up to becoming a superuser.
3. Consider any other service the server has privileged access to as also compromised.
4. Analyse network logs to determine if the server was accessed in an unauthorised way using the compromised key. eg by an unknown IP address.
5. If you are able to confirm no unauthorised access happened. Implement new keys and resume service.
6. Otherwise, destroy all compromised services and restore services on new machines from a backup before the incident occurred. Change all passwords and security keys. Alert users to the extent of the breach.